test/test-bed.js

```diff
@@ -25,44 +24,70 @@ module.exports = class TestBed {
   }
 
   setupTestBed(cwd) {
-    this.rootDir = cwd;
-    this.publicDir = path.join(this.rootDir, 'public');
-    this.buildDir = path.join(this.rootDir, '.greenwood');
-
-    this.teardownTestBed();
-    
-    setupFiles.forEach((file) => {
-      const targetSrc = path.join(process.cwd(), file.dir, file.name);
-      const targetDir = path.join(cwd, file.dir);
-      const targetPath = path.join(cwd, file.dir, file.name);
-
-      if (!fs.existsSync(targetDir)) {
-        fs.mkdirSync(targetDir, { recursive: true });
+    return new Promise(async (resolve, reject) => {
+      try {
+        this.rootDir = cwd;
+        this.publicDir = path.join(this.rootDir, 'public');
+        this.buildDir = path.join(this.rootDir, '.greenwood');
+
+        await this.teardownTestBed();
+
+        await Promise.all(setupFiles.map((file) => {
+          return new Promise(async (resolve, reject) => {
+            try {
+              const targetSrc = path.join(process.cwd(), file.dir, file.name);
+              const targetDir = path.join(cwd, file.dir);
+              const targetPath = path.join(cwd, file.dir, file.name);
+
+              await new Promise(async(resolve, reject) => {
+                try {
+                  await fs.ensureDir(targetDir, { recursive: true });
+                  await fs.copy(targetSrc, targetPath);
+                  resolve();
+                } catch (err) {
+                  reject();
```

Comment:
we should probably pass the `err` to the `reject()`

Reply:
Yes good _catch_

Reply:
> good  _catch_

this guy.... 🤣